src/model/chord.rs

```diff
@@ -0,0 +1,50 @@
+use crate::{Pitch, Quality};
+use combine::{Parser, Stream};
+use std::fmt;
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct Chord {
+    pub pitch: Pitch,
+    pub quality: Quality,
+}
+
+impl Chord {
+    pub fn new(pitch: Pitch, quality: Quality) -> Self {
+        Self { pitch, quality }
+    }
+
+    pub fn parser<Input>() -> impl Parser<Input, Output = Self>
+    where
+        Input: Stream<Token = char>,
+    {
+        (Pitch::parser(), Quality::parser()).map(|(pitch, quality)| Chord::new(pitch, quality))
+    }
+}
+
+impl fmt::Display for Chord {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(&format!("{}{}", self.pitch, self.quality))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use strum::IntoEnumIterator;
+
+    #[test]
+    fn parser() {
+        let chords: Vec<_> = Pitch::all()
+            .into_iter()
```

Comment:
Seeing that `Pitch::all()` itself uses an iterator internally that it then `.collect()`'s into a `Vec`, seems like there'd be an efficiency it letting that "all iterator" be directly consumed? I don't know if there's an idiomatic way to expose that (eg still expose `all()` for convenience when you do want a `Vec`, but also expose `all_iter()` (and probably implement `all()` in terms of `all_iter()`))?

Reply:
Good point. I think just exposing `all()` which returns an iterator is fine. It's easy enough for consumers to collect that into whatever they need. This is another nice use of `impl Trait`!

```rust
pub fn all() -> impl Iterator<Item = Pitch>
```

Reply:
So then can't this `.into_iter()` be removed?